server/src/main/java/org/elasticsearch/cluster/routing/allocation/shards/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -520,18 +540,30 @@ boolean areAllShardsOfThisTypeUnavailable(ShardRouting routing, ClusterState sta
      * (a newly created index having unassigned replicas for example), we don't want the cluster
      * to turn "unhealthy" for the tiny amount of time before the shards are allocated.
      */
-    static boolean isNewlyCreatedAndInitializingReplica(ShardRouting routing, ClusterState state) {
+    static boolean isNewlyCreatedAndInitializingReplica(
+        ShardRouting routing,
+        ClusterState state,
+        TimeValue replicaUnassignedThreshold,
+        Clock clock
```

Comment:
A clock seems like overkill for this, do you think this should instead be passed `Instant.now().toEpochMilli()` and just take a `long`?